ITISFoundation/osparc-simcore
# 🐛 report correct status on user service fail

## What do these changes do?



When a user service fails to start, currently nothing was being reported.
An error will be raised and the service will be shutdown.


All containers which failed to start will be listed inside the `UnexpectedContainerStatusError`
```
ERROR:simcore_service_director_v2.modules.dynamic_sidecar.scheduler.task:Observation of service_name='dy-sidecar_6c93f9f3-d27b-4dd5-a40c-b7a0463dec43'  unexpectedly failed [OEC:139825978785952]
Traceback (most recent call last):
  File "/home/scu/.venv/lib/python3.9/site-packages/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/task.py", line 444, in _observing_single_service
    await apply_observation_cycle(self.app, self, scheduler_data)
  File "/home/scu/.venv/lib/python3.9/site-packages/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_task_utils.py", line 49, in apply_observation_cycle
    await dynamic_scheduler_event.action(app, scheduler_data)
  File "/home/scu/.venv/lib/python3.9/site-packages/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/events.py", line 346, in action
    raise UnexpectedContainerStatusError(
simcore_service_director_v2.modules.dynamic_sidecar.errors.UnexpectedContainerStatusError: Unexpected status from containers: [DockerContainerInspect(status=<Status2.exited: 'exited'>, container_state=ContainerState(Status=<Status2.exited: 'exited'>, Running=False, Paused=False, Restarting=False, OOMKilled=False, Dead=False, Pid=0, ExitCode=123, Error='', StartedAt='2022-11-28T15:09:27.874817258Z', FinishedAt='2022-11-28T15:09:28.40140194Z', Health=None), name='/dy-sidecar-6c93f9f3-d27b-4dd5-a40c-b7a0463dec43-0-jupyter-smash', id='e201468b381bc97c77092c440bae203dcb1fc44f9f3dd4115c260b3f98c621c9')]
```

**NOTE:** I do not expect side effects. After testing VNS Calibrator service (attaching and detaching an input file) it continues to work as expected. @odeimaiz @elisabettai Hope I did not get lucky.


## Related issue/s



- https://github.com/ITISFoundation/osparc-issues/issues/638
- fixes https://github.com/ITISFoundation/osparc-simcore/issues/3171
- fixes https://github.com/ITISFoundation/osparc-simcore/issues/3579
- closes https://github.com/ITISFoundation/osparc-issues/issues/758

## How to test



- open `smash:3.0.8` create a new notebook and add this in the name at the end of the name before the extension ` (2)`
- close the study (it should save)
- next time you open it an the service will result as failed and sidecar will disappear shortly (Test OK)


## Checklist
